source/core/thermal_expansion_model.cc

```diff
@@ -19,6 +19,11 @@
 std::shared_ptr<ThermalExpansionModel>
 ThermalExpansionModel::model_cast(const Parameters::Fluid &fluid_properties)
 {
-  return std::make_shared<ConstantThermalExpansion>(
-    fluid_properties.thermal_expansion);
+  if (fluid_properties.thermal_expansion_model ==
+      Parameters::Fluid::ThermalExpansionModel::phase_change)
+    return std::make_shared<ThermalExpansionPhaseChange>(
+      fluid_properties.phase_change_parameters);
+  else
+    return std::make_shared<ConstantThermalExpansion>(
+      fluid_properties.thermal_expansion);
 }
```

Comment:
Only curiosity : 
Why keeping _ThermalExpansionModel_ class instead of having only _ConstantThermalExpansion_ and _ThermalExpansionPhaseChange_ with inheritance of _PhysicalPropretyModel_ instead of _ThermalExpansionModel._
And having the model_cast condition in the physical_properties_manager.cc file line 79?

Reply:
It's a good idea that will make the code more clear. but I used the method that is currently implemented for all the properties. maybe we will change it to a base class and derived models structure in the future.

Reply:
The idea is that eventually Thermal Expansion, Specific Heat, etc. might have more information in common or might need additional sanity checks that can only be provided if we know what type of properties they are. The idea is to have some sort of interface where at one point you know that aproperty is viscosity for example, otherwise properties will be moved around and could be used in weird fashion (e.g. specific heat passed when it should be a thermal conductivity). I thought it made the interface saner this way.